sport-moment-nft/contracts/EPL.cdc

```diff
@@ -0,0 +1,1124 @@
+            let oldToken <- self.ownedNFTs[id] <- token
+
+            emit Deposit(id: id, to: self.owner?.address)
+
+            destroy oldToken
+        }
+
+        /// batchDeposit takes a Collection object as an argument
+        /// and deposits each contained NFT into this Collection
+        ///
+        pub fun batchDeposit(tokens: @NonFungibleToken.Collection) {
+            // Get an array of the IDs to be deposited
+            let keys = tokens.getIDs()
+
+            // Iterate through the keys in the collection and deposit each one
+            for key in keys {
+                self.deposit(token: <-tokens.withdraw(withdrawID: key))
+            }
+
+            // Destroy the empty Collection
+            destroy tokens
+        }
+
+        /// getIDs returns an array of the IDs that are in the collection
+        ///
+        pub fun getIDs(): [UInt64] {
+            return self.ownedNFTs.keys
+        }
+
+        /// borrowNFT gets a reference to an NFT in the collection
+        //
+        pub fun borrowNFT(id: UInt64): &NonFungibleToken.NFT {
+            return (&self.ownedNFTs[id] as &NonFungibleToken.NFT?)!
+        }
+
+        /// borrowMomentNFT gets a reference to an NFT in the collection
+        ///
+        pub fun borrowMomentNFT(id: UInt64): &EPL.NFT? {
+            if self.ownedNFTs[id] != nil {
+                let ref = (&self.ownedNFTs[id] as auth &NonFungibleToken.NFT?)!
+                return ref as! &EPL.NFT
+            } else {
+                return nil
+            }
+        }
+
+        pub fun borrowViewResolver(id: UInt64): &{MetadataViews.Resolver} {
+            let nft = (&self.ownedNFTs[id] as auth &NonFungibleToken.NFT?)!
+            let eplNFT = nft as! &EPL.NFT
+            return eplNFT as &AnyResource{MetadataViews.Resolver}
+        }
+
+        /// Collection destructor
+        ///
+        destroy() {
+            destroy self.ownedNFTs
+        }
+
+        /// Collection initializer
+        ///
+        init() {
+            self.ownedNFTs <- {}
+        }
+    }
+
+    /// public function that anyone can call to create a new empty collection
+    ///
+    pub fun createEmptyCollection(): @NonFungibleToken.Collection {
+        return <- create Collection()
+    }
+
+    //------------------------------------------------------------
+    // Admin
+    //------------------------------------------------------------
+
+    /// An interface containing the Admin function that allows minting NFTs
+    ///
+    pub resource interface NFTMinter {
+        // Mint a single NFT
+        // The Edition for the given ID must already exist
+        //
+        pub fun mintNFT(editionID: UInt64): @EPL.NFT
+    }
+
+    /// A resource that allows managing metadata and minting NFTs
+    ///
+    pub resource Admin: NFTMinter {
+        /// Borrow a Series
+        ///
+        pub fun borrowSeries(id: UInt64): &EPL.Series {
```

Comment:
Similar to what I said above, I think we should move away from the pattern of borrowing a series/set/play/etc and instead we should just include all the logic to do the things we want in the methods of the Minter resource. That way it makes it more clear what each method does and makes the transactions more static, which is good for FLIX

Reply:
Removed. Thank you.

Reply:
Done. Thank you.